okhttp/src/main/java/okhttp3/internal/http/HttpEngine.java

```diff
@@ -356,17 +293,13 @@ public Connection getConnection() {
    * engine that should be used for the retry if {@code e} is recoverable, or null if the failure is
    * permanent. Requests with a body can only be recovered if the body is buffered.
    */
-  public HttpEngine recover(IOException e, boolean routeException, Sink requestBodyOut) {
+  public HttpEngine recover(IOException e, boolean routeException) {
     streamAllocation.streamFailed(e);
 
     if (!client.retryOnConnectionFailure()) {
       return null; // The application layer has forbidden retries.
     }
 
-    if (requestBodyOut != null && !(requestBodyOut instanceof RetryableSink)) {
```

Comment:
Is this safe to remove? Am thinking about the case of StreamedRequestBody with multiple Routes, and the first route fails.